pkg/scheduler/server/config.go

```diff
@@ -22,22 +22,32 @@ import (
 	"go.etcd.io/etcd/server/v3/embed"
 
 	"github.com/dapr/dapr/pkg/modes"
-	"github.com/dapr/dapr/pkg/security"
 )
 
-func (s *Server) conf() *embed.Config {
+func config(opts Options) (*embed.Config, error) {
+	clientPorts := make(map[string]string)
+	for _, input := range opts.EtcdClientPorts {
+		idAndPort := strings.Split(input, "=")
+		if len(idAndPort) != 2 {
+			return nil, fmt.Errorf("Incorrect format for client ports: %s. Should contain <id>=<client-port>", input)
+		}
+		schedulerID := strings.TrimSpace(idAndPort[0])
+		port := strings.TrimSpace(idAndPort[1])
+		clientPorts[schedulerID] = port
+	}
+
 	config := embed.NewConfig()
 
-	config.Name = s.id
-	config.Dir = s.dataDir + "-" + security.CurrentNamespace() + "-" + s.id
-	config.InitialCluster = strings.Join(s.etcdInitialPeers, ",")
-	config.QuotaBackendBytes = s.etcdSpaceQuota
-	config.AutoCompactionMode = s.etcdCompactionMode
-	config.AutoCompactionRetention = s.etcdCompactionRetention
+	config.QuotaBackendBytes = opts.EtcdSpaceQuota
+	config.AutoCompactionMode = opts.EtcdCompactionMode
+	config.AutoCompactionRetention = opts.EtcdCompactionRetention
+	config.Name = opts.EtcdID
+	config.Dir = opts.DataDir
+	config.InitialCluster = strings.Join(opts.EtcdInitialPeers, ",")
 
-	etcdURL, peerPort, err := peerHostAndPort(s.id, s.etcdInitialPeers)
+	etcdURL, peerPort, err := peerHostAndPort(opts.EtcdID, opts.EtcdInitialPeers)
 	if err != nil {
-		log.Warnf("Invalid format for initial cluster. Make sure to include 'http://' in Scheduler URL")
```

Comment:
can we leave this or move this warning somewhere else? Its really hard to know that you are missing the `http://` but its required and it doesn't exactly err in a nice way

Reply:
Added the string to the return error.